cdap-app-fabric/src/main/java/io/cdap/cdap/app/runtime/ProgramStateWriter.java

```diff
@@ -47,6 +47,16 @@ void start(ProgramRunId programRunId, ProgramOptions programOptions, @Nullable S
    */
   void running(ProgramRunId programRunId, @Nullable String twillRunId);
 
+  /**
+   * Updates the program run's status to be stopping
+   *
+   * @param programRunId the id of the program run
+   * @param gracefulShutdownSecs the time duration in seconds for which we want to wait for the program
```

Comment:
graceful shutdown time duration in seconds to wait for program to be terminated on its own before issuing kill. The value should be Integer.MAX_VALUE if kill shouldnt be issued.